src/fs/mod.rs

```diff
@@ -118,28 +130,53 @@ impl Filesystem {
         let mut buf: Vec<u8> = vec![0; size];
         let mut total = 0;
 
-        'blocks: for block in file.blocks.iter().skip(chunk_index) {
-            let (_, mut fd) = match cache.get(block).await {
-                Ok(out) => out,
-                Err(_) => {
-                    return Ok(req.reply_error(libc::EIO)?);
-                }
+        'blocks: for block in file_metadata.blocks.iter().skip(chunk_index) {
+            // fhash works as a key inside the LRU
+            let fhash = block.hash.clone();
+            let mut lruf = self.lruf.lock().await;
+
+            // getting the file descriptor from the LRU or from the cache if not found in the LRU
+            let mut fd = if lruf.contains(&fhash) {
+                let file_descriptor = lruf.pop(&fhash).unwrap();
+                drop(lruf);
+                file_descriptor
+            } else {
+                let (_, file_descriptor) = match cache.get(block).await {
+                    Ok(out) => out,
+                    Err(_) => {
+                        return Ok(req.reply_error(libc::EIO)?);
+                    }
+                };
+                file_descriptor
             };
+            
+            // seek to the position <offset>
```

Comment:
You still do a seek. We discussed the seek can be optimized if the lru contains struct that owns both the file and it's seek. Of course a good design will create a new File type that can track it's seek internally and only do if necessary.

Reply:
Yes, I let it as it is because the checking for the current position is also calling the seek function internally, so adding the check will consume more CPU time
Also, the file descriptor (fd) has its own pos member internally

Reply:
That's what I meant. I meant you can create a new type that owns both the file object and it's internal seek.